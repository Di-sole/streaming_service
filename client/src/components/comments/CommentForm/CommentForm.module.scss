.container {
    display: flex;
    flex-direction: row;

    &__inputWrapper {
        width: 100%;
        margin-right: 10px;
    }
}

.tooltip {
    margin-top: 5px;
    font-size: 12px;
    color: rgb(189, 0, 0);
}