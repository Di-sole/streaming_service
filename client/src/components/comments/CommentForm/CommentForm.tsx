import InputUI from '@/components/UI/Input/InputUI';
import styles from './CommentForm.module.scss';
import { useRef, useState } from 'react';
import ColoredButton from '@/components/UI/buttons/ColoredButton/ColoredButton';
import { useTranslation } from 'next-i18next';
import { useRouter } from 'next/router';

interface CommentFormProps {
    onSubmit: (value: string) => void
}

const CommentForm = ({ onSubmit }: CommentFormProps) => {
    const { t } = useTranslation('movie');
    const [value, setValue] = useState<string>('');
    const [focus, setFocus] = useState<boolean>(false);

    const inputWrapperRef = useRef(null);

    const clickHandler = (event: React.MouseEvent) => {
        const inputWrapper: HTMLDivElement = inputWrapperRef.current!;

        if (inputWrapper.firstElementChild === event.target) {
            setFocus(true);
        } else {
            setFocus(false);
        }
    };

    return (
        <div className={styles.container}>
            <div
                className={styles.container__inputWrapper}
                ref={inputWrapperRef}
                onClick={(event) => clickHandler(event)}
            >
                <InputUI
                    focus={focus}
                    type="comment"
                    inputType="text"
                    placeholder={t('comments.placeholder')}
                    value={value}
                    onChange={(value) => setValue(value)}
                    onClick={() => setValue('')}
                />
            </div>
            <ColoredButton 
                className={styles.container__button}
                size='large'
                color='gray'
                onClick={() => {
                    onSubmit(value);
                    setValue('')
                }}
            >
                {t('comments.send')}
            </ColoredButton>
        </div>
    );
};

export default CommentForm;
