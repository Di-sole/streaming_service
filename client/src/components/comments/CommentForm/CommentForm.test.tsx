import { render, screen } from '@testing-library/react';
import CommentForm from './CommentForm';

import * as reduxHooks from '../../../store/hooks/redux';

jest.mock('../../../store/hooks/redux');
const useDispatch = jest.spyOn(reduxHooks, 'useAppDispatch');
const mockDispatch = jest.fn();

jest.mock('next/router', () => ({
    useRouter() {
        return {
            route: '/',
            asPath: '/',
            pathname: '',
        };
    },
}));

describe('COMMENTFORM SNAPSHOTS TESTS', () => {
    test('CommentForm should not be changed', () => {
        useDispatch.mockReturnValue(mockDispatch);
        render(<CommentForm onSubmit={() => false} />);

        const id = screen.getByTestId('commentForm');
        expect(id).toMatchSnapshot();
    });
});
