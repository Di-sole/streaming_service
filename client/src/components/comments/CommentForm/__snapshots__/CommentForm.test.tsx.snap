// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMMENTFORM SNAPSHOTS TESTS CommentForm should not be changed 1`] = `
<div
  class="container"
  data-testid="commentForm"
>
  <div
    class="container__inputWrapper"
  >
    <div
      class="container container_comment "
      data-testid="input"
    >
      <div
        class="container__placeholder"
      >
        comments.placeholder
      </div>
      <input
        class="container__input"
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    class="button button_large button_gray container__button"
    type="button"
  >
    comments.send
  </button>
</div>
`;
